lib/bridge/MatrixHandler.js

```diff
@@ -813,6 +813,11 @@ MatrixHandler.prototype.quitUser = Promise.coroutine(function*(req, userId, clie
             }
         }
 
+        if (banned) {
+            // We have already been disconnected.
```

Comment:
can we guarantee this? Not sure how killing a dead connection would impact the state machine (ie: can we just not have this condition?)

Reply:
Yeah, I actually added this line to stop some things from exploding --- we probably don't *actually* need to avoid calling disconnect if it's been called already.

Reply:
fair, just seemed weird to have a thing that stopped termination. If it breaks though, the condition is needed :D